lib/engine/matchers/all_keys_matcher.rb

```diff
@@ -15,7 +15,7 @@ def initialize
     # @param key [string] key value to be matched
     #
     # @return [boolean] true for all instances
-    def match?(_key, _attributes)
+    def match?(_key, _split_treatment, _attributes)
```

Comment:
I think that if this is a different abstraction, its ok to pass an object that evaluates and just returns treatments. Maybe I would rename the parameter, to avoid confusion. split_treatment sounds more like the treatment value, instead of something that evaluates a split and returns a treatment